vueui/README.md

```diff
@@ -0,0 +1,180 @@
+
+To create ui itself, you need to create `.vue` file somewhere in `\vueui\components\view`. Example vueui file:
+```vue
+<template>
+    <div>
+        <p>{{ counter }}</p>
+        <vui-button :params="{ action: 'test', data: 'This is from ui.' }">Call topic</vui-button>
+        <vui-button @click="counter++">Increment counter</vui-button>
+    </div>
+</template>
+
+<script>
+export default {
+  name: 'view-uiname',
+  data() {
+    return this.$root.$data.state; // Make data more easily accessible
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+p {
+    font-size: 3em;
+}
+</style>
+```
+### Step 5: Compile
+This ui framework requires whole ui to be compiled for changes to be available. Compilation requires Node.js runtime, what is obtainable in various ways, most common is install from official site. To do initial dependency setup run `npm install` to gather all dependencies needed for ui. Single compilation can be done with `npm run compile`, but if you constantly do changes, then `npm run run` is more convenient, as it compiles everything as soon as change is detected.
+# Notes
+## Useful APIs
+### `SSvueui.check_uis_for_change(object)`
+Asks all ui's to call `object`'s `vueui_data_change` proc to make all uis up tp date. Should be used when bigger change was done or action done change that would affect global data.
+### `SSvueui.get_open_uis(object)`
+Gets a list of all open uis for specified object. This allows to interact with individual uis. 
+### `ui.activeui`
+Determines currently active view component for this ui datum. Should be combines with `check_for_change()` or `push_change()`. 
+
+It is also plausible for server to send templates dynamically. First character of template should be `?` to specify to vueui that its going to be not recompiled `view` component, but a template. Also template should have single root element. Example for such component would be `?<span>Time since server boot: {{ $root.$data.wtime / 2 }} seconds.</span>`
+### `ui.header`
+Determines header component that is used with this ui. Should be set before `ui.open()`.
+### `ui.auto_update_content`
+Constantly checks for change using `ui.check_for_change()`. This should be only used when data changes unpredictably , 
+### `ui.open()`
+Tries to open this ui, and sends all necessary assets for proper ui rendering. If ui has no data, then calls `object.vueui_data_change(null ...)` to obtain initial ui data.
+### `ui.send_asset(name)`
+Sends singular asset to client for use in ui, after this call you might need to update client-side asset index. To do so call `push_change(null)`.
+### `ui.add_asset(name, image)`
+Adds asset for ui use, but does not send it to client. It will be sent in during next `ui.open()` call or if it's done manually with `ui.send_asset(name)` combined with `push_change(null)`.
+### `ui.remove_asset(name)`
+Removes asset from future use in ui. But client-side asset index isn't updated immediately to reflect removal of asset.
```

Comment:
I don't believe the asset cache supports removing assets. It's also global.

Reply:
This asset part isn't part of asset cache, it's special server generated image asset management proc. Was designed mostly having records in mind.